libzfs.pyx

```diff
@@ -2050,6 +2052,8 @@ cdef class ZFSVdev(object):
         if rv != 0:
             raise self.root.get_error()
 
+        clear_label(self.path)
```

Comment:
If this raises an exception, we will not have written the history entry.

Reply:
Thank you @freqlabs for pointing it out